common/appscale/common/retrying.py

```diff
@@ -0,0 +1,147 @@
+import functools
+import logging
+import random
+import traceback
+import time
+
+DEFAULT_BACKOFF_BASE = 2
+DEFAULT_BACKOFF_MULTIPLIER = 0.2
+DEFAULT_BACKOFF_THRESHOLD = 300
+DEFAULT_MAX_RETRIES = 10
+DEFAULT_RETRYING_TIMEOUT = 60
+DEFAULT_RETRY_ON_EXCEPTION = None   # Retry after any exception.
+
+MISSED = object()
+
+
+class BothMissedException(Exception):
+  pass
+
+
+def not_missed(value_1, value_2):
+  if value_1 is not MISSED:
+    return value_1
+  if value_2 is not MISSED:
+    return value_2
+  raise BothMissedException()
+
+
+class _Retry(object):
```

Comment:
It looks like this could be combined with RetryCoroutine if given an extra coroutine parameter. I don't feel strongly about it though if you'd rather they remain separate.